curriculum/challenges/english/25-front-end-development/lab-lightbox-viewer/66db57ad34c7089b9b41bfd6.md

```diff
@@ -0,0 +1,398 @@
+const targetSrc = 'https://cdn.freecodecamp.org/curriculum/labs/stonehenge-thumbnail.jpg';
+
+for (let image of images) {
+    if (image.src === targetSrc) { 
+        srcCount++; 
+    }
+}
+
+assert(srcCount === 1);
+```
+
+Within the `.gallery` `div`, you should an image element with the `src` set to `https://cdn.freecodecamp.org/curriculum/labs/storm-thumbnail.jpg`.
+
+```js
+const images = document.querySelectorAll('.gallery .gallery-item');
+let srcCount = 0; 
+const targetSrc = 'https://cdn.freecodecamp.org/curriculum/labs/storm-thumbnail.jpg';
+
+for (let image of images) {
+    if (image.src === targetSrc) { 
+        srcCount++; 
+    }
+}
+
+assert(srcCount === 1);
+```
+
+Within the `.gallery` `div`, you should an image element with the `src` set to `https://cdn.freecodecamp.org/curriculum/labs/trees-thumbnail.jpg`.
+
+```js
+const images = document.querySelectorAll('.gallery .gallery-item');
+let srcCount = 0; 
+const targetSrc = 'https://cdn.freecodecamp.org/curriculum/labs/trees-thumbnail.jpg';
+
+for (let image of images) {
+    if (image.src === targetSrc) { 
+        srcCount++; 
+    }
+}
+
+assert(srcCount === 1); 
+```
+
+You should have an element with the class of `lightbox` inside your `body` element.
+
+```js
+// assert either div or span
+assert.exists(document.querySelector('body .lightbox'));
+```
+
+Within your `.lightbox` `div`, you should have a `span` element with the `id` set to `close`.
+
+```js
+assert.equal(document.querySelector('.lightbox #close').tagName, 'SPAN');
+```
+
+Within your `.lightbox` `div`, you should have an `img` element with the `id` set to `lightbox-image`.
+
+```js
+assert.equal(document.querySelector('.lightbox #lightbox-image').tagName, 'IMG');
+```
+
+Your `.lightbox` element should have fixed positioning.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.lightbox')?.position, 'fixed');
+```
+
+Your `.lightbox` element should cover the entire viewport.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.lightbox')?.width, '100%');
+assert.equal(new __helpers.CSSHelp(document).getStyle('.lightbox')?.height, '100%');
+```
+
+Your `.lightbox` element should be aligned with top left corner of the container.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.lightbox')?.top, '0px');
+assert.equal(new __helpers.CSSHelp(document).getStyle('.lightbox')?.left, '0px');
+```
+
+Your `.lightbox` element should have a background color.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.lightbox')?.backgroundColor);
+```
+
+Your `.lightbox` element should be hidden initially.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('.lightbox')?.display, 'none');
+```
+
+When you click one of your `.gallery-item` elements, the `.lightbox` element’s `display` property should be set to `flex` to make it, and the two elements within it, visible.
+
+```js
+// Get the lightbox element
+const lightbox = document.getElementById("lightbox");
```

Comment:
```suggestion
const lightbox = document.querySelector(".lightbox");
```
You should get it by its class because your are not requiring that element to have an `id`.